### PR TITLE
PWX-21857 - El9 compilation fixes

### PR DESCRIPTION
**Pull request by jrivera-px**
Signed-off-by: Jose Rivera <jose@portworx.com>

**What this PR does / why we need it**:
- El9 support 
**Which issue(s) this PR fixes** (optional)
Closes #
PWX-22489 
**Special notes for your reviewer**:

### CODE REVIEW

File: pxd.c [hunk 1]
```diff
@@ -896,6 +900,9 @@ ssize_t pxd_add(struct fuse_conn *fc, struct pxd_add_ext_out *add)
 	struct pxd_device *pxd_dev_itr;
 	int new_minor;
 	int err;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,15,0)
+	int irc;
```
Comment (sulakshm) [inline pxd.c:904]: whats the purpose of this defining this local variable, if this is not checked anywhere.

Comment (jrivera-px) [reply pxd.c:904]: Thanks for looking LNS,  Sorry I should have added a comment.  This actually came up when I was checking these changes against other kernel builds so it failed on "5.16.5-1.el8.elrepo.x86_64"  kernel.  The declaration for `device_add_disk` changed in 5.16 to __must_check...   If you know a better way to handle this please let me know?
 I tried "(void)device_add_disk" and that did not work
KERNELPATH=/usr/src/kernels/5.16.5-1.el8.elrepo.x86_64 make  clean all
...
/usr/src/px-fuse-rc7/pxd.c: In function ‘pxd_add’:
/usr/src/px-fuse-rc7/pxd.c:973:9: error: ignoring return value of ‘device_add_disk’ declared with attribute ‘warn_unused_result’ [-Werror=unused-result]
  973 |         device_add_disk(&pxd_dev->dev, pxd_dev->disk, NULL);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

**Comment by sulakshm** [comment]
Simple, add check for return code failure and fail the call.

**Comment by jrivera-px** [comment]
👍

**Review by prabirpaul — APPROVED** [review]
looks good.

**Review by prabirpaul — APPROVED** [review]

**Pull request merged**